File: src/main/java/threads/ThreadGeneralBehaviour.java
package threads;


import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
import utils.MessageUtils;

public class ThreadGeneralBehaviour implements Runnable {

    MessageReceivedEvent event;
    String message;

    public ThreadGeneralBehaviour(MessageReceivedEvent event) {
	this.event = event;
    }

    @Override
    public void run() {
	message = event.getMessage().getContentDisplay();

	// Let's go for matching all possible cases
	// We don't use else if since we want to match multiple cases

	if (message.matches("(.*(?i)yousoro.*)|(.*sylux6Yo.*)")) {
	    if (!MessageUtils.reactMessage(event.getMessage(), "yousoro")) {
	    	// Blue heart
			event.getMessage().addReaction("\uD83D\uDC99").queue();
	    }
	}
	
	if (message.matches("(.*\\W+|)((?i)zensoku+ +zenshi+n+)(\\W+.*|)")) {
	    MessageUtils.sendMessage(event.getChannel(), "YOUSORO!~ (> ᴗ •)ゞ");
	}
	
	else if (message.matches("(.*\\W+|)((?i)best +waifu)(\\W+.*|)")) {
	    MessageUtils.sendMessage(event.getChannel(), "わたし？");
	}
	
	else if (message.matches("(?i).*hello.*|.*ohayo.*|.*good +morning.*")) {
	    MessageUtils.sendMessage(event.getChannel(), MessageUtils.mentionAt(event.getAuthor()) + " Ohayousoro! (> ᴗ •)ゞ");
	}
	
    }

}

File: src/main/kotlin/com/github/sylux6/watanabot/threads/ThreadGeneralBehaviour.kt
package com.github.sylux6.watanabot.threads

import com.github.sylux6.watanabot.utils.message.mentionAt
import com.github.sylux6.watanabot.utils.message.reactMessage
import com.github.sylux6.watanabot.utils.message.sendMessage
import net.dv8tion.jda.api.events.message.MessageReceivedEvent

open class ThreadGeneralBehaviour(internal var event: MessageReceivedEvent) : Runnable {
    internal lateinit var message: String

    override fun run() {
        message = event.message.contentDisplay

        // Let's go for matching all possible cases
        // We don't use else if since we want to match multiple cases

        if (message.matches("(.*(?i)yousoro.*)|(.*sylux6Yo.*)".toRegex())) {
            if (!reactMessage(event.message, "yousoro")) {
                // Blue heart
                event.message.addReaction("\uD83D\uDC99").queue()
            }
        }

        when {
            message.matches("(.*\\W+|)((?i)zensoku+ +zenshi+n+)(\\W+.*|)".toRegex()) -> sendMessage(
                event.channel,
                "YOUSORO!~ (> ᴗ •)ゞ"
            )
            message.matches("(.*\\W+|)((?i)best +waifu)(\\W+.*|)".toRegex()) -> sendMessage(
                event.channel,
                "わたし？"
            )
            message.matches("(?i).*hello.*|.*ohayo.*|.*good +morning.*".toRegex()) -> sendMessage(
                event.channel,
                mentionAt(event.author) + " Ohayousoro! (> ᴗ •)ゞ"
            )
        }
    }
}
